### PR TITLE
Build result from runner

### PR DESCRIPTION
**Pull request by nex-54**
Need ability to get buildId on post buildQueue #71

### Checklist
- [x] All unit tests passed on build server
- [x] Acceptance test(s) covers new/modified functionality 
- [ ] Code coverage is at least the same or higher
- [x] Breaking change in public API

var connection = new RemoteTc().Connect(_ => _.ToHost("tc"));
build = connection.RunBuildConfiguration(having => having.Id("bt2"));
Thread.Sleep(10000);
var status = connection.GetBuild(build.Id).Status;

### CODE REVIEW

File: FluentTc.Tests/AcceptanceTests.cs [hunk 1]
```diff
@@ -221,13 +221,13 @@ public void RunBuildConfiguration_WithComment()
             var connectedTc = new RemoteTc().Connect(_ => _.AsGuest(), teamCityCaller, buildConfigurationRetriever);
 
             // Act
-            connectedTc.RunBuildConfiguration(having, options => options.WithComment("comment!"));
+            var build = connectedTc.RunBuildConfiguration(having, options => options.WithComment("comment!"));
```
Comment (borismod) [inline FluentTc.Tests/AcceptanceTests.cs]: @nex-54 that's great that RunBuildConfiguration has a return value. Now you can assert on it.

File: FluentTc.Tests/AcceptanceTests.cs [hunk 2]
```diff
@@ -221,13 +221,13 @@ public void RunBuildConfiguration_WithComment()
...
 
             // Assert
             A.CallTo(
                 () =>
-                    teamCityCaller.Post("<build>\r\n<buildType id=\"bt2\"/>\r\n<comment><text>comment!</text></comment>\r\n</build>\r\n",
-                    HttpContentTypes.ApplicationXml, "/app/rest/buildQueue", ""))
+                    teamCityCaller.PostFormat<BuildModel>("<build>\r\n<buildType id=\"bt2\"/>\r\n<comment><text>comment!</text></comment>\r\n</build>\r\n",
```
Comment (borismod) [inline FluentTc.Tests/AcceptanceTests.cs]: @nex-54 can you please assert on the returned value of build?

File: FluentTc.Tests/AcceptanceTests.cs [hunk 3]
```diff
@@ -442,17 +442,42 @@ public void RunBuildConfiguration_OnAgentNameWithParameters()
...
+            // Arrange
+            Action<IBuildConfigurationHavingBuilder> having = _ => _.Name("FluentTc");
+            var teamCityCaller = CreateTeamCityCaller();
+            var buildConfigurationRetriever = A.Fake<IBuildConfigurationRetriever>();
+
+            A.CallTo(() => buildConfigurationRetriever.GetSingleBuildConfiguration(having))
+                .Returns(new BuildConfiguration { Id = "bt2" });
+            A.CallTo(
+                () =>
+                    teamCityCaller.PostFormat<BuildModel>(A<string>.Ignored,
```
Comment (borismod) [inline FluentTc.Tests/AcceptanceTests.cs]: @nex-54 can you replace "A<string>.Ignored" with the actual string that you expect it to be?

**Comment by borismod** [comment]
@nex-54 I like this contribution very much. I always prefer methods that return value rather than do something with a "side-effect". Methods with return values are mush easier to test. Can you please do another step and address a few comments I left in the code?

**Comment by nex-54** [comment]
All done except "percentage of statement coverage" -  i'm not really sure what should i do to increase percentage of statement coverage. All my changes is covered by tests.

**Comment by borismod** [comment]
@nex-54 thank you for your contribution and I really appreciate your effort and patience to improve the coverage. I'll accept your PR and will add some tests later on to improve the coverage. Thanks again and welcome to the FluentTc contributors!

**Comment by borismod** [comment]
closes #71

**Pull request merged**